assistant_slash_commands: Fix AI text thread path display bugs on Windows and all platforms

CLoaKY233 opened the pull request:
## Fix incorrect directory path folding in slash command file collection

**Description:**
This PR fixes a bug in the `collect_files` function where the directory folding logic (used to compact chains like `.github/workflows`) failed to reset its state when traversing out of a folded branch.

**The Issue:**
The `folded_directory_names` accumulator was persisting across loop iterations. If the traversal moved from a folded directory (e.g., `.github/workflows`) to a sibling directory (e.g., `.zed`), the sibling would incorrectly inherit the prefix of the previously folded path, resulting in incorrect paths like `.github/.zed`.

**The Fix:**
*   Introduced `folded_directory_path` to track the specific path currently being folded.
*   Added a check to reset `folded_directory_names` whenever the traversal encounters an entry that is not a child of the currently folded path.
*   Ensured state is cleared immediately after a folded directory is rendered.

**Release Notes:**
- Fixed an issue where using slash commands to collect files would sometimes display incorrect directory paths (e.g., showing `.github/.zed` instead of `.zed`) when adjacent directories were automatically folded.

dinocosta commented on crates/assistant_slash_commands/src/file_command.rs:
```diff
@@ -301,19 +300,22 @@ fn collect_files(
                         };
+                        if is_top_level_directory {
+                            is_top_level_directory = false;
+                        }
```
I believe we can revert back to the original code here, that way we avoid checking `is_top_level_directory` twice and avoid the `filename.clone()` call 👍

CLoaKY233 replied:
Done! :D Sorry for sloppy code !

dinocosta commented on crates/assistant_slash_commands/src/file_command.rs:
```diff
@@ -301,19 +300,22 @@ fn collect_files(
                         events_tx.unbounded_send(Ok(SlashCommandEvent::Content(
                             SlashCommandContent::Text {
-                                text: label.to_string(),
+                                // text: path_including_worktree_name.display(path_style).to_string(),
```
I believe we can remove this comment? 🤔

```suggestion
```

CLoaKY233 commented on crates/assistant_slash_commands/src/file_command.rs:
```diff
@@ -289,8 +289,7 @@ fn collect_files(
                                 folded_directory_names =
                                     folded_directory_names.join(&path_including_worktree_name);
                             } else {
-                                folded_directory_names =
```
That's weird, 
- for me, it's working properly for this build,
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/48a1bb94-06f9-4b33-8d2e-23c186484e36" />

- should this be platform specific?

CLoaKY233 commented:
Thanks for the thoughtful review and for investigating the Windows issues,
If there's anything specific I can help with (testing, logs, etc.), I'll be up for it.

Veykril committed:
assistant_slash_commands: Fix text threads not listing files on windows correctly

CLoaKY233 committed:
Track and reset folded directory path

Add folded_directory_path to remember the currently folded directory and
clear it when leaving that path, skipping empty directories, or after
rendering a directory to prevent fold state leaking to unrelated
entries.

Veykril commented:
I don't think its supposed to be like that, but thats a bug in the current release already, lets not mix that in here.

Veykril commented:
If you can come up with a test for that would be appreciated! I asked someone else internally to double check this change as I am not too familiar with text threads (my concern was the windows bug here)
